### PR TITLE
Reduce the amount of amount of python2

### PR DESCRIPTION
**Pull request by grimreaper**
Also stop recommending that people run pip as root and instead just use
virtualenv.

### CODE REVIEW

File: api/examples/python/README.md [hunk 1]
```diff
@@ -8,16 +8,13 @@ notifications.
 First, install the dependencies.
 
 ```
-sudo /usr/bin/python -m pip install websocket-client
-sudo /usr/bin/python -m pip install protobuf
-
+mkvirtualenv iterm2api
```
Comment (gnachman) [inline api/examples/python/README.md]: What's mkvirtualenv? I don't have it by default on my system. I agree sudo is awful, but I don't want to add even more dependencies on third party packages.

Comment (grimreaper) [inline api/examples/python/README.md]: mkvirtualenv is a fairly common wrapper around python's virtualenv tools: 
https://virtualenvwrapper.readthedocs.io/en/latest/command_ref.html

You can get the same effect with
```
mkdir -p ~/.virtualenv
cd ~/.virtualenv
virtualenv xxx
source ./xxx/bin/activate
```

Comment (gnachman) [inline api/examples/python/README.md]: OK to add a section describing how to do it using virtualenv as an alternative to `sudo python` but I don't want to insist on more deps than are necessary. Dependencies are the root of all frustration.

Comment (grimreaper) [inline api/examples/python/README.md]: IMHO modifying system python using sudo is more likely to lead to eventual frustration than using isolated environments, but I'll rewrite this section to be an alternative

**Comment by Cabalist** [comment]
Hey there,
I support this pull request but it would be worthwhile to remove the space after `print` to clarify that it is a function (PEP8 \*cough\*). 

Also as the is example code I don't think creating a `main()` function really adds anything.   It is totally fine to put it under `if __name__...`

My 2¢ as a random python dev...

**Comment by grimreaper** [comment]
> it would be worthwhile to remove the space after print to clarify that it is a function (PEP8 *cough*).

done.

> It is totally fine to put it under if __name__...
Personally I always use a main() even if writing simple scripts.

**Comment by mennis** [comment]
virtenv is an additional dependency both in terms of code and conceptually.  The --user option to the pip install command will install a package for the current user, rather than for all users.

**Comment by gnachman** [comment]
@mennis Does that mean you wouldn't need to use sudo? How should the doc be changed to use `--user`?

**Comment by mennis** [comment]
Yes.   `pip install --user  websocket-client protobuf ` will work without sudo. In my case it will install the packages in ${HOME}/Library/Python/2.7/lib/python/site-packages/ .

**Comment by grimreaper** [comment]
Its generally best practice to not install project dependencies either as sudo or as user. They should be in a virtualenv. The user of --user meant for binaries that you expect to be on the PATH.

**Pull request merged**